src/measures/fermions/stag.hpp

```diff
@@ -59,19 +59,30 @@ namespace nissa
     delete[] NAME2(loc,o)
 #define PRINT(A)							\
       master_fprintf(file,"%+16.16lg %+16.16lg\t",A[0]/meas_pars.nhits,A[1]/meas_pars.nhits)
+#define PRINT_VEC(A)							\
+      master_fprintf(file," # %s flav %d\n",#A,iflav);    \
+      for(int t=0;t<glbSize[0];t++)                     \
+      master_fprintf(file,"%+16.16lg %+16.16lg\t",A[t][0]/meas_pars.nhits,A[t][1]/meas_pars.nhits);         \
```

Comment:
here, a complex is expected... maybe you could write two routines `print_vec_impl` (implementation), one taking a double vector and one taking a complex

Reply:
of course one could have a print_vec routine by itself....

Reply:
but we only need the one for the double case... So  I'm gonna create two routines `print_vec_complex` and `print_vec_double` anyway, and at the place of `PRINT_VEC` in `SUMM_THE_TIME_TRACE_PRINT_AT_LAST_HIT` we're gonna call  `print_vec_double`, right?

Reply:
you mean that PRINT_VEC is used only in this situation, I see. So you can directly assume it's a real vector without the need to tell apart the case of complex. If needed in the future we can write the case of a complex

Reply:
should I pass to the `print_vec_double` the generic size for the array one passes or just one argument for our specific case?

Reply:
if PRINT_VEC is not used anywhere else, you can adapt the macro as you want